### PR TITLE
tests: Use GTEST_SKIP

### PR DESCRIPTION
**Pull request by sjfricke**
Just more cleanup to use `GTEST_SKIP` when possible

### CODE REVIEW

File: tests/vklayertests_dynamic_rendering.cpp [hunk 1]
```diff
@@ -1787,21 +1774,14 @@ TEST_F(VkLayerTest, TestFragmentDensityMapRenderArea) {
     }
 
     if (!DeviceExtensionSupported(gpu(), nullptr, VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME)) {
-        printf("%s %s not supported, skipping test\n", kSkipPrefix, VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME);
-        return;
-    }
-
-    if (!DeviceExtensionSupported(gpu(), nullptr, VK_EXT_FRAGMENT_DENSITY_MAP_EXTENSION_NAME)) {
-        printf("%s %s not supported, skipping test\n", kSkipPrefix, VK_EXT_FRAGMENT_DENSITY_MAP_EXTENSION_NAME);
-        return;
```
Comment (ncesario-lunarg) [inline tests/vklayertests_dynamic_rendering.cpp:1796]: This test is failing in our CI with:
```
[ RUN      ] VkLayerTest.TestFragmentDensityMapRenderArea
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-VkRenderingInfo-pNext-06112'
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-VkRenderingInfo-pNext-06114'
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-VkRenderingInfo-pNext-06113'
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-VkRenderingInfo-pNext-06115'
[  FAILED  ] VkLayerTest.TestFragmentDensityMapRenderArea (572 ms)
```

Is this check still needed?

File: tests/vklayertests_dynamic_rendering.cpp [hunk 2]
```diff
@@ -1869,18 +1849,11 @@ TEST_F(VkLayerTest, TestFragmentDensityMapRenderAreaWithoutDeviceGroupExt) {
...
```
Comment (ncesario-lunarg) [inline tests/vklayertests_dynamic_rendering.cpp:1883]: This test is also failing:
```
[ RUN      ] VkLayerTest.TestFragmentDensityMapRenderAreaWithoutDeviceGroupExt
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-VkRenderingInfo-imageView-06110'
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-VkRenderingInfo-imageView-06111'
[  FAILED  ] VkLayerTest.TestFragmentDensityMapRenderAreaWithoutDeviceGroupExt (673 ms)
```

**Review by ncesario-lunarg — COMMENTED** [review]
Some failing tests, possibly due to some checks that were removed. LGTM once that's resolved and passes CI.

File: tests/vklayertests_dynamic_rendering.cpp (hunk 1 above)
Comment (sjfricke) [reply tests/vklayertests_dynamic_rendering.cpp:1796]: this should be `!AreRequestedExtensionsEnabled()`, seems this slipped by

**Pull request merged**